dts/arm/ambiq/ambiq_apollo4p.dtsi

```diff
@@ -25,6 +25,10 @@
 		compatible = "mmio-sram";
 		reg = <0x10000000 0x60000>;
 	};
+
+	pinctrl: pin-controller {
```

Comment:
why is this a _virtual device_? if not device specific (as Nordic), it should have a register

Reply:
Added base address to the pinctrl node.